test/reservation_manager_test.rb

```diff
@@ -0,0 +1,228 @@
+require_relative 'test_helper'
+
+describe "Reservation Manager Class" do 
+	before do 
+		@sample = HotelManager::ReservationManager.new()
+	end
+
+	it "is an instance of Reservation Manager" do
+		expect(@sample).must_be_kind_of HotelManager::ReservationManager
+		expect(@sample.rooms).must_be_kind_of Array
+		expect(@sample.reservation_blocks).must_be_kind_of Array
+	end
+
+	it "confirm that hotel contains 20 rooms" do
+		expect(@sample.rooms.length).must_equal 20
+	end
+
+	it "able to list out all rooms in the hotel" do
+		expect(@sample.rooms_list()).must_be_kind_of String
+		expect(@sample.rooms_list()).must_include "Room: 1, Room: 2, Room: 3"
+		expect(@sample.rooms_list()).must_include "Room: 19, Room: 20"
+	end
+end
+
+describe "Reservation Manager Class - Allow Booking" do
+	before do 
+		@sample = HotelManager::ReservationManager.new()
```

Comment:
It is good to use meaningful variable names to increase readability. It's not immediately clear what `@sample` us. Consider `@hotel`.